src/MongoDB.Driver.Core/Core/Servers/ServerFactory.cs

```diff
@@ -25,16 +25,24 @@ namespace MongoDB.Driver.Core.Servers
     internal class ServerFactory : IClusterableServerFactory
     {
         // fields
+#pragma warning disable CS0618
         private readonly ClusterConnectionMode _clusterConnectionMode;
+        private readonly ConnectionModeSwitch _connectionModeSwitch;
+#pragma warning restore CS0618
         private readonly IConnectionPoolFactory _connectionPoolFactory;
+        private readonly bool? _directConnection;
         private readonly IServerMonitorFactory _serverMonitorFactory;
         private readonly IEventSubscriber _eventSubscriber;
         private readonly ServerSettings _settings;
 
         // constructors
-        public ServerFactory(ClusterConnectionMode clusterConnectionMode, ServerSettings settings, IConnectionPoolFactory connectionPoolFactory, IServerMonitorFactory serverMonitoryFactory, IEventSubscriber eventSubscriber)
+#pragma warning disable CS0618
+        public ServerFactory(ConnectionModeSwitch connectionModeSwitch, ClusterConnectionMode clusterConnectionMode, bool? directConnection, ServerSettings settings, IConnectionPoolFactory connectionPoolFactory, IServerMonitorFactory serverMonitoryFactory, IEventSubscriber eventSubscriber)
```

Comment:
Use standard order of parameters:

    public ServerFactory(ClusterConnectionMode clusterConnectionMode, ConnectionModeSwitch connectionModeSwitch, bool? directConnection, ServerSettings settings, IConnectionPoolFactory connectionPoolFactory, IServerMonitorFactory serverMonitoryFactory, IEventSubscriber eventSubscriber)